Lib/Core/IdRegisterCore.pas

```diff
@@ -137,7 +137,6 @@ implementation
   IdGlobal,
   {$ENDIF}
 
-  IdBaseComponent,

```

Comment:
`TIdBaseComponent` is being used by `TIdBaseComponentSelectionEditor` in Delphi, so this unit needs to be `IFDEF`'ed on `HAS_TSelectionEditor`.